File: backend/app/api/v1/site.py
"""站点配置公开接口。

- GET /site-config  前端启动配置（帖子图片开关等；数据库不可用时返回默认值）
"""

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.db import get_db
from app.services import site_config as site_config_service

router = APIRouter(prefix="/site-config", tags=["站点配置"])


@router.get("", summary="获取前端公开站点配置")
async def get_site_config(session: AsyncSession = Depends(get_db)) -> dict:
    return await site_config_service.get_public_config(session)

File: frontend/src/api/site.ts
import http from './http'

/** 站点公开配置（管理后台可配置，见 /admin/config/site）。 */
export interface SiteConfig {
  site_name?: string
  post_image_enabled?: boolean
}

export function getSiteConfig() {
  return http.get<unknown, SiteConfig>('/site-config')
}
